### PR TITLE
[CHAT] Refactor ChatGPTProvider to use `openai-java` client

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

- use Java SDK `openai-java` for ChatGPT which is popular and listed in official website, https://github.com/TheoKanning/openai-java
- Focus on lifecycle in ChatGPTProvider, and prevent handling lower-level concepts in details, like POJO mapping, HTTP request handling.
- follow the changes from upstream changes from OpenAI

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.readthedocs.io/en/master/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
Please check the following things:

- it introduces new deps into Kyuubi binary tarball, thus LICENSE-binary and NOTICE-binary should be updated
- the backend of retrofit should be okhttp, please make sure it does not create no-daemon threads which may block engine shutdown.

File: externals/kyuubi-chat-engine/src/main/scala/org/apache/kyuubi/engine/chat/provider/ChatGPTProvider.scala [hunk 1]
```diff
@@ -75,29 +74,19 @@ class ChatGPTProvider(conf: KyuubiConf) extends ChatProvider {
...
-        val replyMessage = mapper.treeToValue[Message](
-          respJson.get("choices").get(0).get("message"))
-        messages.addLast(replyMessage)
-        replyMessage.content
-      case errorStatusCode =>
+    try {
+      messages.addLast(new ChatMessage("user", q))
+      val completionRequest = ChatCompletionRequest.builder()
+        .messages(messages.asScala.toList.asJava)
+        .model("gpt-3.5-turbo")
```
Comment (pan3793) [inline externals/kyuubi-chat-engine/src/main/scala/org/apache/kyuubi/engine/chat/provider/ChatGPTProvider.scala:81]: let's make it configurable in next PR :)

**Comment by pan3793** [comment]
I remember that okhttp provides a mock server module for testing, we can add UT then.

File: externals/kyuubi-chat-engine/src/main/scala/org/apache/kyuubi/engine/chat/provider/ChatGPTProvider.scala (hunk 1 above)
Comment (bowenliang123) [reply externals/kyuubi-chat-engine/src/main/scala/org/apache/kyuubi/engine/chat/provider/ChatGPTProvider.scala:81]: Sure. Make it configurable of other ChatCompletionRequest settings like model name, max tokens, etc.

**Comment by bowenliang123** [comment]
> Please check the following things:
> 
> * it introduces new deps into Kyuubi binary tarball, thus LICENSE-binary and NOTICE-binary should be updated
> * the backend of retrofit should be okhttp, please make sure it does not create no-daemon threads which may block engine shutdown.

How can I update the LICENSE-binary and  NOTICE-binary, in manual way or auto-collect by tools?

**Comment by bowenliang123** [comment]
> I remember that okhttp provides a mock server module for testing, we can add UT then.

Shall we do the ut in the next PR ?

**Comment by pan3793** [comment]
> How can I update the LICENSE-binary and NOTICE-binary, in manual way or auto-collect by tools?

Currently, we should update them manually, we haven't found a tool smart enough to automatically process them.

**Comment by pan3793** [comment]
> > I remember that okhttp provides a mock server module for testing, we can add UT then.
> 
> Shall we do the ut in the next PR ?

it's fine.

**Comment by bowenliang123** [comment]
image
Locally tested with the packaged tarball. Works properly as before refactoring, with
- message history in the context of session
- HTTP proxy

**Comment by bowenliang123** [comment]
Thanks, merged to master.

**Pull request closed without merging**